### PR TITLE
SFR-623 update reader visuals for SFR 

### PR DESCRIPTION
**Pull request by hokei**
This is a very initial pass to get reader visuals updated for SFR's beta.  
- Changed pagination bars into buttons 
- Top navigation doesn't auto-hide
- Modified top navigation, settings and TOC labels 
- Bottom is always chapter navigation
- Modified iframe width/height handling so that only the main scrollbar appears
- "Fixed" (hacked-together) bug where focus gets lost when tabbing into iframe

### CODE REVIEW

**Comment by hokei** [comment]
@mwbenowitz I still need to fix the tests, but due to time it will probably be commenting out old tests more than actually writing too much new stuff.  I hope I can get to it sometime today, but it's still worth poking around in the meantime

File: examples/streamed/misc/readers/reader-NYPL/webpub-viewer.js [hunk 1]
```diff
@@ -637,26 +638,55 @@ define("ColumnsPaginatedBookView", ["require", "exports", "HTMLUtilities", "Brow
             this.hasFixedScrollWidth = (body.scrollWidth === originalScrollWidth);
             body.style.left = originalLeft + "px";
         };
+        // Get available width for iframe container to sit within
```
Comment (mwbenowitz) [inline examples/streamed/misc/readers/reader-NYPL/webpub-viewer.js:641]: This seems to work well on larger screens, but leaves no margin on mobile devices. I know that is a low priority for the moment, but want to make a note of it (if one hasn't already been made)

File: examples/streamed/misc/readers/reader-NYPL/webpub-viewer.js [hunk 2]
```diff
@@ -637,26 +638,55 @@ define("ColumnsPaginatedBookView", ["require", "exports", "HTMLUtilities", "Brow
...
+                var bottomRect = bottomBar.getBoundingClientRect();
+                bottomHeight = bottomRect.height;
+            }
+            return window.innerHeight - topHeight - bottomHeight;
+        };
         ColumnsPaginatedBookView.prototype.setSize = function () {
             // any is necessary because CSSStyleDeclaration type does not include
             // all the vendor-prefixed attributes.
             var body = HTMLUtilities.findRequiredIframeElement(this.bookElement.contentDocument, "body");
-            var width = (BrowserUtilities.getWidth() - this.sideMargin * 2) + "px";
```
Comment (mwbenowitz) [inline examples/streamed/misc/readers/reader-NYPL/webpub-viewer.js:644]: Related to above, I guess removing this `sideMargin` is what's doing it?

File: examples/streamed/misc/readers/reader-NYPL/webpub-viewer.js [hunk 3]
```diff
@@ -1291,18 +1319,44 @@ define("ScrollingBookView", ["require", "exports", "BrowserUtilities", "HTMLUtil
...
+            }
+            var nextBtn = document.getElementById('next-page-btn');
+            var nextBtnWidth = 0;
+            if (nextBtn) {
+                var rect = nextBtn.getBoundingClientRect();
+                nextBtnWidth = rect.width;
+            }
+            return window.innerWidth - prevBtnWidth - nextBtnWidth;
+        };
+        ScrollingBookView.prototype.getAvailableHeight = function () {
```
Comment (mwbenowitz) [inline examples/streamed/misc/readers/reader-NYPL/webpub-viewer.js:1339]: Hooray for scrolling mode!

**Review by mwbenowitz — APPROVED** [review]
This looks really good! The functionality is much better than what was there before and using it feels much more natural. There are a few minor things that I want to raise to address later, I don't think any of them should block this merge or need to be addressed immediately. They are:

- The margin issue on mobile/small screens
- The page navigation arrows still appear and behave oddly when the table of contents is open. They should probably just be hidden I think
- The previous/next chapter links don't seem to know when the book is at the first/last chapter and should probably be inactive in those cases.

**Comment by hokei** [comment]
Yes!  I'll make tickets for these things, thank you for taking a look!

**Pull request merged**